vendor/engines/secure_rooms/app/controllers/secure_rooms_api/events_controller.rb

```diff
@@ -2,11 +2,37 @@ module SecureRoomsApi
 
   class EventsController < SecureRoomsApi::ApiController
 
+    before_action :parse_time
+
     def create
-      Rails.logger.info "SecureRoomsApi Event: #{params}"
+      SecureRooms::Alert.create(alert_params)
```

Comment:
We could possibly do this as something like (I haven't actually verified this is syntactically correct):
```
SecureRooms::Alert.create(
  alert_params.merge(
    message_time: parse_time(params[:message_time)),
    raw: JSON.parse(request.body),
  )
)
```

I do it occasionally, but I generally prefer to avoid modifying the `params` hash.